src/bootstrap/config.rs

```diff
@@ -965,6 +965,9 @@ impl Config {
         config.changelog_seen = toml.changelog_seen;
 
         let build = toml.build.unwrap_or_default();
+        if let Some(file_build) = build.build {
+            config.build = TargetSelection::from_user(&file_build);
+        };
```

Comment:
NB build triple wasn't read from result of `get_toml`. I don't know whether it was skipped deliberately or just simply forgotten. I'm leaning towards the latter, since `build.host` and `build.target` are read below from the same toml.

Reply:
probably accidental; https://github.com/rust-lang/rust/blob/8adbe6e436d7ae03dbb0e82729745e1e514eb508/src/bootstrap/config.rs#L815 gets it right 99% of the time and python sets it for https://github.com/rust-lang/rust/blob/8adbe6e436d7ae03dbb0e82729745e1e514eb508/src/bootstrap/build.rs#L4-L6 the rest of the time in https://github.com/rust-lang/rust/blob/8adbe6e436d7ae03dbb0e82729745e1e514eb508/src/bootstrap/bootstrap.py#L788-L797, so likely no one ever noticed.

I *think* it shouldn't ever matter in practice; even when we stop using python (https://github.com/rust-lang/rust/issues/94829) the shell scripts should still respect `build.build` in config.toml, but changing it to make things testable seems good 👍